[Term Entry] Matplotlib hist

caupolicandiaz opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

A term entry for the Matplotlib .hist() method.

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [ x] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

SSwiniarski reviewed (approved):
@caupolicandiaz, Looks good. I don't have anything to add.

yangc95 commented on content/matplotlib/concepts/pyplot/terms/hist/hist.md:
````diff
@@ -0,0 +1,85 @@
+matplotlib.pyplot.hist(x, bins, range, density, weights, cumulative, bottom, histtype, align, orientation, rwidth, log, color, label, stacked)
+```
+
+The `x` argument, the data to be plotted, is the only required parameter. Other parameters are optional and can be used to customize plot elements such as the number of bins, color, and other relevant features.
+
+`.hist()` takes the following arguments:
+
+- `x`: Values to be plotted of type float or array.
````
```suggestion
The `x` argument is the only required parameter. It represents the values that will be plotted and can be of type float or array.

Other parameters are optional and can be used to customize plot elements such as the number of bins, color, and other relevant features.

`.hist()` takes the following optional arguments:

```

yangc95 commented on content/matplotlib/concepts/pyplot/terms/hist/hist.md:
```diff
@@ -0,0 +1,85 @@
+- `weights`: Accepts an array of values (must match the shape of `x`) that determines the relative contribution of each value in `x`.
+- `cumulative`: Takes a boolean value, and if `True` computes a cumulative histogram where each bin gives counts of the relevant values within that bin in addition to all the values smaller than that bin.
+- `bottom`: Sets the floor or the bottom of the bins, default is 0.
+- `histtype`: Takes keywords (e.g. 'bar', 'barstacked', 'step', 'stepfilled') to modify the histogram type.
+- `align`: Sets the alignment of the bins, default is 'mid' (options include 'left' and 'right').
```
Can we put strings in backticks

yangc95 reviewed (commented):
Just had a few comments @caupolicandiaz

caupolicandiaz commented:
> Just had a few comments @caupolicandiaz

I updated the entry based on your edits and I also updated the example. I wasn't happy with the original, now the code is a little more streamlined and I think the graph is simpler, and cleaner.

yangc95 commented on content/matplotlib/concepts/pyplot/terms/hist/hist.md:
```diff
@@ -0,0 +1,82 @@
+
+# Function for standardizing pet age values
+def get_months(x):
+    if x.endswith('years'):
+        return abs(int(x.split(' ')[0]) * 12) # Some year entries included negative values
+    elif x.endswith('weeks'):
+        return math.ceil(int(x.split(' ')[0]) / 4 )
+    else:
+        return int(x.split(' ')[0])
```
Would it be possible to set up two variables here? One for number (the int that's extracted) and another for the unit? In addition, are there some strings with no units?

caupolicandiaz replied:
This is supposed to be a quick and dirty function to create a new column, to "explore" the data. It's not really meant as a function for reuse. The data being addressed is a bit of a mess, it includes three different units, negative values and NaNs. I did update the last line because I hadn't noticed the "days" entries (so I'm rounding any such entry up to 1 month).

yangc95 replied:
I see, did we want to keep an `elif` for the case of month(s)? I think this is good to go after

caupolicandiaz replied:
I bungled the edit. When I reviewed the data again I also saw that some values are not plural (1 month vs 3 months), so I redesigned the function again. Should be all set.

yangc95 commented:
@caupolicandiaz would you mind opening a quick typo PR and change the media link to begin with "https://raw.githubusercontent.com/Codecademy/docs/main/"

yangc95 commented:
live link: https://www.codecademy.com/resources/docs/matplotlib/pyplot/hist
